### PR TITLE
bpo-29972: Skip tests known to fail on AIX

### PR DESCRIPTION
**Pull request by vstinner**
See http://bugs.python.org/issue29972 for more information.

### CODE REVIEW

**Comment by vstinner** [comment]
@edelsohn: Can you please try this change on AIX? (I'm unable to test my change.)

**Comment by vstinner** [comment]
(Oops, I forgot some "import sys", it should now be fixed.)

**Comment by edelsohn** [comment]
With the patches,

test_eintr
test_locale
test_strptime

succeed.

test_socket fails with
======================================================================
FAIL: testIPv6toString (test.test_socket.GeneralModuleTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch/dje/src/cpython/Lib/test/test_socket.py", line 1090, in testIPv6toString
    assertInvalid('1:2:3:4:5:6:7:8:')
  File "/scratch/dje/src/cpython/Lib/test/test_socket.py", line 1066, in <lambda>
    (OSError, ValueError), f, a
AssertionError: (<class 'OSError'>, <class 'ValueError'>) not raised by <lambda>

----------------------------------------------------------------------
with a similar guard for the '1:2:3:4:5:6:7:8:' test, it succeeds.

test_tools fails because test_tools/test_i18n.py needs to import sys.  With that change, it succeeds.

File: Lib/test/test_socket.py [hunk 1]
```diff
@@ -1081,7 +1083,9 @@ def testIPv6toString(self):
         assertInvalid('::0::')
         assertInvalid('1::abc::')
         assertInvalid('1::abc::def')
-        assertInvalid('1:2:3:4:5:6:')
+        # bpo-29972: inet_pton() doesn't fail on AIX
+        if not sys.platform.startswith('aix'):
+            assertInvalid('1:2:3:4:5:6:')
         assertInvalid('1:2:3:4:5:6')
         assertInvalid('1:2:3:4:5:6:7:8:')
```
Comment (edelsohn) [inline Lib/test/test_socket.py]: '1:2:3:4:5:6:7:8:' needs to be guarded on AIX also.

**Comment by vstinner** [comment]
I already added "import sys" to test_tools/test_i18n.py, maybe you picked the old version of my PR?

> '1:2:3:4:5:6:7:8:' needs to be guarded on AIX also.

Ok, done. Can you please test again?

**Comment by edelsohn** [comment]
Yes, if with that additional one skipped, it passes.

Thanks David

**Comment by vstinner** [comment]
Ok, I merged this first changes to fix (skip) a few unit tests on the AIX buildbot.

**Pull request merged**